weldx/asdf/tags/weldx/core/geometry/point_cloud.py

```diff
@@ -29,7 +31,9 @@ def to_tree(cls, node: PointCloud, ctx):
             'weldx.core.ExternalFile' type to be serialized.
 
         """
-        return node.__dict__
+        tree = deepcopy(node.__dict__)
```

Comment:
just out of curiosity, does asdf modify the passed dict?

Reply:
Yes. It replaces all python objects in the dict with a dictionary representation of this object. For some reason (I am not totally aware of its purpose) the to tree method gets called twice when you serialize an object. However, in this case, it's really helpful since it usually creates an error during the second execution when the dictionary of a dataclass gets overwritten.

Reply:
calling it twice for the same instance sounds like a bug to me.

Reply:
Well, as far as I remember, it is not. For some reason, they had to do it twice. I just can't remember the specifics anymore. I think it had something to do with some things that couldn't be evaluated during the first pass.